File: domain/src/main/java/com/everis/domain/interactor/CreateUser.java
package com.everis.domain.interactor;

import com.everis.domain.executor.PostExecutorThread;
import com.everis.domain.repository.UserRepository;

import rx.Observable;

/**
 * Created by everis on 25/04/18.
 */

public class CreateUser extends UseCase {

    private UserRepository repository;
    private String email;
    private String password;

    public CreateUser(PostExecutorThread postExecutorThread, UserRepository repository) {
        super(postExecutorThread);
        this.repository = repository;
    }

    public void bindParams(String email, String password){
        this.email = email;
        this.password = password;
    }

    @Override
    protected Observable buildUseCaseObservable() {
        return repository.createUser(this.email, this.password);
    }
}

File: domain/src/main/java/com/everis/domain/interactor/CreateUser.kt
package com.everis.domain.interactor

import com.everis.domain.executor.PostExecutorThread
import com.everis.domain.model.User
import com.everis.domain.repository.UserRepository

import rx.Observable

class CreateUser(postExecutorThread: PostExecutorThread, private val repository: UserRepository) : UseCase<User>(postExecutorThread) {
    private var email: String? = null
    private var password: String? = null

    fun bindParams(email: String, password: String) {
        this.email = email
        this.password = password
    }

    override fun buildUseCaseObservable(): Observable<User> {
        return repository.createUser(this.email!!, this.password!!)
    }
}
